README.md

```diff
@@ -1,80 +1,84 @@
-This is the start state that is generated when the relation is joined.
-A charm layer responding to this state should get the common name, a list of 
-Subject Alt Names, and the certificate_name call 
-`request_server_cert(common_name, sans, certificate_name)` on the relation 
-object.
+```yaml
+requires:
+  cert-provider:
+    interface: tls-certificates
+```
 
-## {relation_name}.ca.available
-The Certificate Authority is available on the relation object when the 
-"{relation_name}.ca.available" state is set. The charm layer can retrieve the
-CA by calling `get_ca()` method on the relationship object.
+Next, you must ensure the interface layer is included in your `layer.yaml`:
 
-```python
-from charms.reactive import when
-@when('certificates.ca.available')
-def store_ca(tls):
-    certificate_authority = tls.get_ca()
+```yaml
+includes:
+  - interface:tls-certificates
 ```
 
-## {relation_name}.server.cert.available
-Once the server certificate is set on the relation the interface layer will
-emit the "{relation_name}.server.cert.available" state, indicating that the 
-server certificate is available from the relationship object.  The charm layer 
-can retrieve the certificate and use it in the code by calling the
-`get_server_cert()` method on the relationship object.
+Then, in your reactive code, add the following, changing `update_certs` to
+handle the certificates however your charm needs:
 
 ```python
-from charms.reactive import when
-@when('certificates.server.cert.available')
-def get_server(tls):
-    server_cert, server_key = tls.get_server_cert()
-```
+from charmhelpers.core import hookenv, host
+from charms.reactive import endpoint_from_flag
 
-## {relation_name}.client.cert.available
-Once the client certificate is set on the relation the interface layer will
-emit the "{relation_name}.client.cert.available" state, indicated that the
-server certificates is available from the relationship object.  The charm layer
-can retrieve the certificate and use it in the code by calling the
-`get_client_cert()` method on the relationship object.
 
-```python
-from charms.reactive import when
-@when('certificates.client.cert.available')
-def store_client(tls):
-    client_cert, client_key = tls.get_client_cert()
+@when('cert-provider.ca.changed')
+def install_root_ca_cert():
+    cert_provider = endpoint_from_flag('cert-provider.ca.available')
+    host.install_ca_cert(cert_provider.root_ca_cert)
+    clear_flag('cert-provider.ca.changed')
+
+
+@when('cert-provider.available')
+def request_certificates():
+    cert_provider = endpoint_from_flag('cert-provider.available')
+
+    # use first ingress address as primary and any additional as SANs
+    client_ingress = hookenv.network_get('clients')['ingress-addresses']
+    cert_provider.request_server_cert(client_ingress[0], client_ingress[:1])
+
+    db_ingress = hookenv.network_get('db')['ingress-addresses']
+    cert_provider.request_client_cert(db_ingress[0], db_ingress[:1])
```

Comment:
I assume request_client_cert is idempotent, but should there be mention of that here?

Reply:
It is idempotent in that requests are sorted by their common name value and requests with the same common name will replace a previous request with the same common name.  In theory, you could have two certs with the same common name, but there wasn't really a way to handle that in a backwards compatible way, and I don't think there's been any real need for it thus far.

I agree that this should be called out in the docs, so I'll update that.